advanced/src/hashTable.js

```diff
@@ -0,0 +1,199 @@
+function Node(key, value) {
+  this.key = key
+  this.value = value
+  this.next = null
+}
+
+export default class LinkedList {
+  constructor() {
+    this.length = 0
+    this.head = null
+  }
+
+  clear() {
+    this.length = 0
+    this.head = null
+  }
+
+  insert(key, value) {
+    let node = new Node(key, value)
+    let currentNode = this.head
+
+    if (!currentNode) {
+      this.head = node
+      this.length++
+      return node
+    }
+    while (currentNode.next) {
+      currentNode = currentNode.next
+    }
+    currentNode.next = node
+    this.length += 1
+    return node
+  }
+
+//regular find
+  find(key) {
+    var currentNode = this.head
+    if (!currentNode) {
+      return null
+    }
+    while (currentNode) {
+      if (currentNode.key === key){
+        return currentNode
+      } else {
+        currentNode = currentNode.next
+      }
+    } return -1
+  }
+
+//finds the node whose next value is the target node
+  findByNext(key) {
+    let currentNode = this.head
+
+    if(!currentNode) {
+      return null
+    } else if (!currentNode.next && currentNode.key === key){
+      return clear()
+    } else {
+      while(currentNode.next){
+        if(currentNode.next.key === key){
+          return currentNode
+        } else {
+          currentNode = currentNode.next
+        }
+      } return null
+    }
+  }
+
+//cleaner remove using findByNext
+  removeWithFindByNext(key) {
```

Comment:
i like that you are utilizing two versions is the remove. there is a little code smell here. but, in the interest of learning it's valid. You may eventually want to come up with a `remove()` function alone